File: miroulis.py
#This program reads a file of numbers
#and prints the minimum and maximum values
#Written by Andrew Miroulis
#Written in Python
#Replit compiler
#4/21/23

def main():
  f = open("readnum.txt", "r")
  lines = f.readlines()
  nums = []
  
  #For loop iterates through each line
  #and appends the nums array with every
  #line without the new line character
  for line in lines:
    if line[-1] == '\n':
      nums.append(line[:-1])
    else:
      nums.append(line)
  min = int(nums[0])
  max = int(nums[0])
  
  #Loop iterates through nums and compares
  #each number to min and max and updates
  #either value accordingly
  for num in nums:
    if int(num) < min:
      min = int(num)
    elif int(num) > max:
      max = int(num)

  print("Min: " + str(min) + "\n")
  print("Max: " + str(max) + "\n")
  
  f.close()
main()

File: miroulis.rb
#This program takes 5 prime numbers between 0 and 50 
#from the user and calculates the sum
#If not all numbers are prime, the program
#prints an error message
#If any number is not in the specified range,
#the program will print an error message
#Written in Ruby
#Written by Andrew Miroulis
#Replit Compiler
#4/28/23


nums = [5]
print("Enter 5 prime numbers between 0 and 50 \n")
nums[0] = gets.chomp.to_i
nums[1] = gets.chomp.to_i
nums[2] = gets.chomp.to_i
nums[3] = gets.chomp.to_i
nums[4] = gets.chomp.to_i
prime = true
inRange = true
sum = 0
for num in nums
  if(num < 0 or num > 50)
    inRange = false
    print("Error: 1 or more numbers are not between 0 and 50")
    break
  end
  if (num == 1)
    prime = false
  end
  for i in 2..num/2
    if num % i == 0 or num == 1
      prime = false
      break
    end
  end
  if not prime
    print("Error: 1 or more numbers are not prime")
    break
  end
  sum += num
end
if prime and inRange
  print("Sum = " + sum.to_s)
end
  

